### ISSUE TITLE
[k in keyof EnumA] should show error

### ISSUE THREAD

**Issue by bluelovers** [microsoft/TypeScript#31620]
**TypeScript Version:**  3.4.0-dev.201xxxxx

**Search Terms:**

**Code**

```ts
export const enum EnumA
{
	a,
	b,
}

export type Interface = {
	[k in keyof EnumA]: (a: number, b: string) => void
	// here should show error, because it is enum
}

let a: Interface = {
	a(a, b)
	{
		console.log(a.toFixed(), b.blink());
	},
	b(a, b)
	{
		console.log(a.toFixed(), b.blink());
	},
}
```

**Expected behavior:**

> error, and tell use `[k in keyof typeof EnumA]`

**Actual behavior:**

> no error

**Playground Link:** 

**Related Issues:**

**Comment by jcalz** [comment]
I kind of agree with this suggestion... `keyof` on a known primitive type is probably a mistake, right?   How often do people do that intentionally?

**Comment by RyanCavanaugh** [comment]
I'm curious
